icu4c/source/test/intltest/alphaindextst.cpp

```diff
@@ -756,4 +757,52 @@ void AlphabeticIndexTest::checkHasBuckets(const Locale &locale, UScriptCode scri
             uscript_getScript(bucket->getLabel().char32At(0), errorCode));
 }
 
+void AlphabeticIndexTest::checkMaxLabelCount() {
+    IcuTestErrorCode errorCode(*this, "checkMaxLabelCount");
+    UErrorCode status = U_ZERO_ERROR;
+    Locale locale("en");
+    UnicodeString loc = locale.getName();
+    AlphabeticIndex aindex(locale, status);
+
+    int32_t defaultMax = aindex.getMaxLabelCount();
+    assertEquals("Expect max count = 99", 99, defaultMax);
+
+    aindex.setMaxLabelCount(237, status);
+    if (U_FAILURE(errorCode)) {
+      dataerrln("%s %d  Error in index setMaxLabelCount",  __FILE__, __LINE__);
+      return;
+    }
```

Comment:
Why do you use assertEquals() above but not here?

If you do want it like this, then use errln() here not dataerrln().

dataerrln() is appropriate when a failure at that point might have to do with data loading.
Please check the `status` right after the constructor, and use the `UBool errDataIfFailureAndReset(const char *fmt, ...);` method on the errorCode.